### PR TITLE
Add some additional bech32(m) testing

### PR DESCRIPTION
**Pull request by instagibbs**
Testing that the outputs that are created match expected, and added a unit test for invalid addresses from the BIP

### CODE REVIEW

File: common/test/run-param.c [hunk 1]
```diff
@@ -465,6 +457,43 @@ static void deprecated_rename(void)
 		      NULL));
 }
 
+static void invalid_bech32m(void)
+{
+	int wit_version;
+	uint8_t data_out[500];
+    size_t data_out_len;
```
Comment (rustyrussell) [inline common/test/run-param.c:464]: Weird indent?

**Review by rustyrussell — APPROVED** [review]
Ack b653a8044a49f06538bd431586495d1c97cc1329

File: common/test/run-param.c (hunk 1 above)
Comment (instagibbs) [reply common/test/run-param.c]: always one of those, fixed

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5997/commits/691c94ae982cb2fe2ea467f658d1b549fb3142aa

**Comment by instagibbs** [comment]
I put a commit in to log bitcoind logs to stdout, hoping to diagnose why bitcoind seems to be crashing on restart sometimes e.g. https://github.com/bitcoin/bitcoin/issues/27088

**Comment by rustyrussell** [comment]
> I put a commit in to log bitcoind logs to stdout, hoping to diagnose why bitcoind seems to be crashing on restart sometimes e.g. [bitcoin/bitcoin#27088](https://github.com/bitcoin/bitcoin/issues/27088)

Are we deleting the directory underneath it?

**Comment by rustyrussell** [comment]
Trivial rebase to kick CI.

Ack 8e994e542

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5997/commits/8e994e5421ca9c2a75245766ce6c524ee8f9e049

**Pull request merged**